src/fluree/db/transact.cljc

```diff
@@ -205,10 +207,21 @@
 
 (defn publish-commit
   "Publishes commit to all nameservices registered with the ledger."
-  [{:keys [primary-publisher secondary-publishers] :as _ledger} commit-jsonld]
+  [{:keys [primary-publisher secondary-publishers alias] :as _ledger} commit-jsonld]
   (go-try
-    (let [result (<? (nameservice/publish primary-publisher commit-jsonld))]
-      (nameservice/publish-to-all commit-jsonld secondary-publishers)
+    (let [existing-record (when primary-publisher
+                            (<? (nameservice/lookup primary-publisher alias)))
+          branch-metadata (when existing-record
+                            {:created-at (get existing-record "f:createdAt")
+                             :created-from (get existing-record "f:createdFrom")
+                             :protected (get existing-record "f:protected")
+                             :description (get existing-record "f:description")})
+          ;; Add branch metadata to commit if it exists
+          commit-with-metadata (if branch-metadata
+                                 (assoc commit-jsonld "branchMetadata" branch-metadata)
```

Comment:
It feels weird to add information to the commit that isn't actually persisted on the commit - do we need to do this when we publish, since it's already going to live on the ns-record?